lms/services/blackboard_api/client.py

```diff
@@ -31,7 +31,10 @@ def list_files(self, course_id):
         """Return the list of files in the given course."""
 
         files = []
-        path = f"courses/uuid:{course_id}/resources?type=file&limit={PAGINATION_LIMIT}"
+        path = f"courses/uuid:{course_id}/resources"
+        path += "?type=file"
+        path += f"&limit={PAGINATION_LIMIT}"
+        path += "&fields=id,name,modified,mimeType"
```

Comment:
I wonder if this is getting on to where we don't want to be doing this with string concatenation but use something like `from urllib.parse import urlencode` instead. I think it means we're more likely to get encoding issues right and also increases the readability.

```suggestion
        path = f"courses/uuid:{course_id}/resources?" + urlencode({
            "type": "file",
            "limit": PAGINATION_LIMIT,
            "fields": "id,name,modified,mimeType"
        })
```

This would result in `type=file&limit=10&fields=id%2Cname%2Cmodified%2CmimeType`, instead of `type=file&limit=10&fields=id,name,modified,mimeType` which might actually be correct, and a good reason to do this.

If Blackboard is weird and doesn't understand it, then you can add `safe=","` to the call.

Also you can do this as well:

```python
print(urlencode({
    "type": "file",
    "limit": PAGINATION_LIMIT,
    "fields": ["id", "name", "modified", "mimeType"]
}, doseq=True))
```

Which results in `type=file&limit=10&fields=id&fields=name&fields=modified&fields=mimeType` which _should_ be the same, but who knows if Blackboard would accept it.

Reply:
:+1: Done. I didn't try the `doseq` thing because even if it does work it's not documented in Blackboard's docs. Best to stick to what their docs say